### PR TITLE
Add back dir property in spans in text layer

### LINKED ISSUE #12909: Text layer doesn't use dir property anymore
Some characters in:
[hebrew.pdf](https://github.com/mozilla/pdf.js/files/5869562/hebrew.pdf)
are not correct in the text layer (such as question marks).

We get text direction in `evaluator.js` but we don't use it.
It was used in the past, but it seems it has been removed:
https://github.com/mozilla/pdf.js/commit/5bd8a83c9bd7c23003b09e1cfd71422c6b083826#diff-579f2fb8055052aa97906d5fc75aeee84aa7dd9f8efcca24d8a5460d828e1de8L170

### PR DESCRIPTION
**Pull request by calixteman**
- aims to fix #12909

### CODE REVIEW

File: test/pdfs/.gitignore [hunk 1]
```diff
@@ -70,6 +70,7 @@
 !bug1650302_reduced.pdf
 !issue8570.pdf
 !issue8697.pdf
+!issue12909.pdf
```
Comment (Snuffleupagus) [inline test/pdfs/.gitignore:73]: Shouldn't this be a linked test-case instead?

File: test/test_manifest.json [hunk 2]
```diff
@@ -4526,6 +4526,12 @@
        "link": false,
        "type": "eq"
     },
+    {  "id": "issue12909",
+       "file": "pdfs/issue12909.pdf",
+       "md5": "0faa1cb029c3323c168595cccbe46ec0",
+       "rounds": 1,
```
Comment (Snuffleupagus) [inline test/test_manifest.json:4532]: Do you need to test all pages, or can we limit it to just *one* page?

**Review by Snuffleupagus — CHANGES_REQUESTED** [review]

File: test/pdfs/.gitignore (hunk 1 above)
Comment (calixteman) [reply test/pdfs/.gitignore:73]: I don't know.
How do you decide this in general ?

Comment (brendandahl) [reply test/pdfs/.gitignore:73]: I don't think we have a rule, but generally I follow:
- reduced test case - include in repo
- small file author has given us permission to use - include in repo
- linked that will likely keep working (e.g. github, bugzilla) - linked test case
- linked file from random site - try to create reduced test case or use way back machine

Comment (Snuffleupagus) [inline test/pdfs/.gitignore:73]: Please remove this line, since you're now adding a linked test.

**Review by Snuffleupagus — APPROVED** [review]
Looks good to me, with the final comment addressed *and* test run on the Windows-bot as well (just to be safe).

**Comment by calixteman** [comment]
I see that in the log:
`WARNING: MD5 of file "pdfs/issue12909.pdf" does not match file. Expected "0faa1cb029c3323c168595cccbe46ec0" computed "d41d8cd98f00b204e9800998ecf8427e"`
I checked the md5 and it's ok:
```bash
wget "https://web.archive.org/web/20210126084513/https://www.machonso.org/uploads/images/%D7%95%D7%95%D7%9C%D7%A3.pdf" -qO - | md5sum
0faa1cb029c3323c168595cccbe46ec0  -
```
and I get the second md5 with an empty file:
```bash
touch foo && md5sum foo
d41d8cd98f00b204e9800998ecf8427e  foo
```
Could it explain why regression tests aren't ran ?

**Comment by Snuffleupagus** [comment]
Given the previous errors in https://github.com/mozilla/pdf.js/pull/12910#issuecomment-767472177, I'd suggest removing the `issue12909.pdf` *and* `issue12909.pdf.error` files from the Windows bot and trying again.

**Comment by Snuffleupagus** [comment]
Huh, what happened to the fonts in the `eq`-tests?

**Review by timvandermeij — APPROVED** [review]

**Comment by timvandermeij** [comment]
Thank you for fixing this!

/botio makeref

**Pull request merged**